builder/FindGlobals.cmake

```diff
@@ -36,25 +36,26 @@ if( Linux OR Darwin )
 
   include( GNUInstallDirs )
 
-  add_definitions(-DUNIX)
+  append( "-DUNIX" CMAKE_FLAGS )
 
   if( Linux )
-    add_definitions(-D__USE_LARGEFILE64 -D_FILE_OFFSET_BITS=64)
+    append( "-D__USE_LARGEFILE64" CMAKE_FLAGS )
+    append( "-D_FILE_OFFSET_BITS=64" CMAKE_FLAGS )
 
-    add_definitions(-DLINUX)
-    add_definitions(-DLINUX32)
+    append( "-DLINUX" CMAKE_FLAGS )
+    append( "-DLINUX32" CMAKE_FLAGS )
```

Comment:
we should have used compiler standard macro defs instead like `__linux__`, but this will be separate exercise, don't do that within this PR